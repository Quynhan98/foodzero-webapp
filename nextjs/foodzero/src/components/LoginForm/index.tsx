import { ChangeEvent, FormEvent, memo } from 'react'
import {
  FormControl,
  FormLabel,
  Input,
  FormErrorMessage,
  Button,
} from '@chakra-ui/react'

interface LoginFormProps {
  onSubmitForm?: (e: FormEvent<HTMLFormElement>) => void
  onChangeForm?: (e: ChangeEvent<HTMLInputElement>) => void
  emailError?: string
  passwordError?: string
  isDisable: boolean
}

const LoginForm = ({
  onSubmitForm,
  onChangeForm,
  emailError,
  passwordError,
  isDisable,
}: LoginFormProps) => {
  return (
    <form onSubmit={onSubmitForm}>
      <FormControl
        display="flex"
        flexDirection="column"
        justifyContent="center"
        isInvalid={!!(emailError || passwordError)}
        padding={{ base: '25px', md: '50px' }}
      >
        <FormLabel
          fontSize={{ base: 'md', md: 'lg' }}
          fontWeight="bold"
          fontFamily="Rufina"
          textAlign="center"
          pb={{ base: '58px', md: '115px' }}
        >
          Login
        </FormLabel>
        <Input
          data-testid="email"
          type="email"
          name="email"
          placeholder="Email"
          onChange={onChangeForm}
        />
        {emailError && (
          <FormErrorMessage fontSize={{ base: 'xxs', md: 'base' }}>
            {emailError}
          </FormErrorMessage>
        )}
        <Input
          data-testid="password"
          type="password"
          name="password"
          placeholder="Password"
          onChange={onChangeForm}
          mt={{ base: '25px', md: '50px' }}
        />
        {passwordError && (
          <FormErrorMessage fontSize={{ base: 'xxs', md: 'base' }}>
            {passwordError}
          </FormErrorMessage>
        )}
        <Button
          mt={{ base: '40px', md: '80px' }}
          type="submit"
          isDisabled={isDisable}
          _hover={{ backgroundColor: 'none' }}
        >
          Sign In
        </Button>
      </FormControl>
    </form>
  )
}

export default memo(LoginForm)
