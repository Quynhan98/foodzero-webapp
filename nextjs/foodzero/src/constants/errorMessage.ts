export const REQUIRED = 'This field is required.'
export const EMAIL_NOT_EXIST = 'Email address does not exist.'
export const INVALID_EMAIL_FORMAT = 'Invalid email format.'
export const INVALID_PASSWORD_FORMAT = 'Invalid password format.'
export const INVALID_PASSWORD = 'Invalid password.'
export const SERVER_ERROR = 'Something went wrong. Please try again!'
