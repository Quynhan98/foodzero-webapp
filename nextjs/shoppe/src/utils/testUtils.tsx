import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>{component}</ChakraProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
