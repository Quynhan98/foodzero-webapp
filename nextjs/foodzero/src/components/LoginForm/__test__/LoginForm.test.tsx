import React from 'react'

// Utils
import { render, screen, cleanup, fireEvent } from '@utils/testUtils'

// Components
import LoginForm from '@components/LoginForm'
import { INVALID_EMAIL_OR_PASSWORD } from '@constants/errorMessage'

describe('LoginForm', () => {
  const accountMock = {
    email: 'foodzero@gmail.com',
    password: '123456789',
  }

  const props = {
    onChangeForm: jest.fn(),
    onSubmitForm: jest.fn(),
    error: '',
    isDisable: false,
  }

  beforeEach(() => {
    render(<LoginForm {...props} />)
  })

  afterEach(() => {
    cleanup()
  })

  it('LoginForm matches snapshot', () => {
    const { container } = render(<LoginForm {...props} />)

    expect(container).toMatchSnapshot()
  })

  it('Should call onChange input email', () => {
    const email = screen.getByTestId('email')

    fireEvent.change(email, {
      target: { value: accountMock.email },
    })

    expect(props.onChangeForm).toBeCalled()
    expect(email).toHaveValue(accountMock.email)
  })

  it('Should call onChange input password', () => {
    const password = screen.getByTestId('password')

    fireEvent.change(password, {
      target: { value: accountMock.password },
    })

    expect(props.onChangeForm).toBeCalled()
    expect(password).toHaveValue(accountMock.password)
  })

  it('Should call submit form', () => {
    const button = screen.getByRole('button')

    fireEvent.submit(button)

    expect(props.onSubmitForm).toBeCalled()
  })

  it('Should render ErrorMessage', () => {
    render(<LoginForm {...props} error={INVALID_EMAIL_OR_PASSWORD} />)

    expect(screen.getByText(INVALID_EMAIL_OR_PASSWORD)).toHaveTextContent(
      INVALID_EMAIL_OR_PASSWORD,
    )
  })
})
