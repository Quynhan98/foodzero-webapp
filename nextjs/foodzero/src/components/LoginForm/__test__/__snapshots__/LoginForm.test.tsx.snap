// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm LoginForm matches snapshot 1`] = `
<div>
  <form>
    <div
      class="chakra-form-control css-1tu1hsb"
      role="group"
    >
      <label
        class="chakra-form__label css-1e23bqj"
        for="field-:r1:"
        id="field-:r1:-label"
      >
        Login
      </label>
      <input
        class="chakra-input css-pfxawx"
        data-testid="email"
        id="field-:r1:"
        name="email"
        placeholder="Email"
        type="email"
      />
      <input
        class="chakra-input css-wijn7f"
        data-testid="password"
        id="field-:r1:"
        name="password"
        placeholder="Password"
        type="password"
      />
      <button
        class="chakra-button css-rvf1lq"
        type="submit"
      >
        Sign In
      </button>
    </div>
  </form>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
